docs/README.md

```diff
@@ -658,22 +658,244 @@ In this chapter you'll walk through these concepts and its details.
+@Command({
+  authorize: 'all',
+})
+export class CreateProduct {
+  public constructor(
+    readonly sku: string,
+    readonly price: number
+  ) {}
+
+  public async handle(register: Register): Promise<void> {
+    const priceLimit = 10
+    if (this.price >= priceLimit) {
+      throw new Error(`price must be below ${priceLimit}, and it was ${this.price}`)
+    }
+  }
+}
+
+```
+
+And this mutation:
+
+```graphql
+mutation($input: CreateProductInput!) {
+  CreateProduct(input: $input)
+}
+
+# Variables
+
+{
+  "input": {
+    "sku": "MYSKU",
+    "price": 19.99
+  }
+}
+```
+
+You'll get something like this response:
+
+```grapqhl
+{
+  "errors": [
+    {
+      "message": "price must be below 10, and it was 19.99",
+      "path": [
+        "CreateProduct"
+      ]
+    }
+  ]
+}
+```
+
+###### Register error events 
+
+There could be situations in which you want to register an event representing an error. For example, when moving items with insufficient stock from one location to another:
+
+```typescript
+@Command({
+  authorize: [Admin],
+})
+export class MoveStock {
+  public constructor(
+    readonly productID: string,
+    readonly origin: string,
+    readonly destination: string,
+    readonly quantity: number
+  ) {}
+
+  public async handle(register: Register): Promise<void> {
+    if (!this.enoughStock(this.productID, this.origin, this.quantity)) {
+      register.events(new ErrorEvent(`There is not enough stock for ${this.productID} at ${this.origin}`))
+    } else {
+      register.events(new StockMoved(/*...*/))
+    }
+  }
+
+  private enoughStock(productID: string, origin: string, quantity: number): boolean {
+    /* ... */
+  }
+}
+```
+
+In this case, the client who submitted the command can still complete the operation. Then an event handler will take care of that `ErrorEvent` and proceed accordingly.
+
 ##### Reading entities
 
+Event handlers are a good place to make decisions, and for making better decisions, you need information. There is a Booster function called `fetchEntitySnapshots` within the `Booster` package and allows you to inspect the application state. This function receives two arguments, the `Entity` to fetch and the `entityID`. Here is an example of fetching an entity called `Stock`:
+
+```typescript
+@Command({
+  authorize: [Admin],
+})
+export class MoveStock {
+  public constructor(
+    readonly productID: string,
+    readonly origin: string,
+    readonly destination: string,
+    readonly quantity: number
+  ) {}
+
+  public async handle(register: Register): Promise<void> {
+    const stock = await Booster.fetchEntitySnapshot(Stock, this.productID)
+    if (!this.enoughStock(this.origin, this.quantity, stock)) {
+      register.events(new ErrorEvent(`There is not enough stock for ${this.productID} at ${this.origin}`))
+    }
+  }
+
+  private enoughStock(origin: string, quantity: number, stock?: Stock): boolean {
+    return stock !== undefined && stock.countByLocation[origin] >= quantity
```

Comment:
`countByLocation[origin]` could also be `undefined`, perhaps something like this?

```suggestion
  const count = stock?.countByLocation[origin]
  if (count) {
    return count >= quantity
  }
  return false
```

Reply:
to the Nick's suggestion, I would add `count > 1` because if count was -1 the condition would still be true

Reply:
or event something like

`const hasStock = stock?.countByLocation[origin] > 0`

Reply:
Sure @NickSeagull @tainguyenbui thank you for the feedback, we can use the `!!` idiom to keep the snippet short

```typescript
  private enoughStock(origin: string, quantity: number, stock?: Stock): boolean {
    const count = stock?.countByLocation[origin]
    return !!count && count >= quantity
  }
```

Take into account how we calculate the stock is not that important here as how we fetch and use entities within a command handler. It is important to pay attention to these details but it is also to save extra efforts to our readers.

Let me know your thoughts 👍

Reply:
What does `!!` do? Is there a documentation page for it?

Reply:
LOL :) that's a typical ruby pattern to make a boolean expression from something falsy or truly, and it seems it is being used in TypeScript too https://stackoverflow.com/a/49792111